### DISCUSSION TITLE
Full Facial and Eye tracking via HTC Vive Pro Eye and Facial Tracker

### DISCUSSION THREAD

**Discussion by benaclejames** [vrchat-community/osc#100] [Show and tell]
[VRCFaceTracking](https://github.com/benaclejames/VRCFaceTracking)

Created in-depth parameter blending logic, custom "binary" data types to workaround network lag and parameter limit, in-depth documentation/tutorials for all parameters and an active [Discord community](https://discord.gg/Fh4FNehzKn) of around 500 members based around facial tracking within VRChat.

The app also supports external tracking libraries too, so future hardware can be supported via community-made sdk wrappers

## Examples
https://twitter.com/Benaclejames/status/1494864147351281671
https://twitter.com/Benaclejames/status/1496561035213426699
https://twitter.com/kentrl_z/status/1497020472046800897
https://twitter.com/Demirramon/status/1497374424542195712
https://twitter.com/aroma_great/status/1499078678159917056
https://twitter.com/sig_serg/status/1498090076705800199
https://www.youtube.com/watch?v=dMmK-QJWrkw
https://www.youtube.com/watch?v=BaKHjCWNsbk
https://twitter.com/RolltheredDev/status/1498363224114225154
https://www.youtube.com/watch?v=LizBjQGn0_M
https://youtu.be/3s26ncS-Za8
https://cdn.discordapp.com/attachments/881975159429398569/945439186125725787/Face_Tracking.mp4
https://cdn.discordapp.com/attachments/881975159429398569/942945681138679848/2022-02-14_17-39-31.mp4
https://cdn.discordapp.com/attachments/881975159429398569/942175168162312212/Protogen_VRCFT.mp4
**And much more wonderful user-created content to be discovered!**

**Comment by momo-the-monster** [comment]
Thanks for sharing this, @benaclejames ! It's been added to the [list](https://github.com/vrchat-community/osc/wiki/Resources#face-tracking).
